lib/Sema/ResilienceDiagnostics.cpp

```diff
@@ -276,13 +276,16 @@ static bool diagnoseValueDeclRefExportability(SourceLoc loc, const ValueDecl *D,
   }
 
   // Access levels from imports are reported with the others access levels.
-  // Except for extensions, we report them here.
+  // Except for extensions and protocol conformances, we report them here.
   if (originKind == DisallowedOriginKind::InternalOrLessImport ||
       originKind == DisallowedOriginKind::PackageImport) {
-    if (reason != ExportabilityReason::ExtensionWithPublicMembers &&
-        reason != ExportabilityReason::ExtensionWithPackageMembers &&
-        reason != ExportabilityReason::ExtensionWithConditionalConformances &&
-        reason != ExportabilityReason::ExtensionWithPackageConditionalConformances)
+    bool reportHere =
```

Comment:
Would it maybe make sense to use a `switch` to cover `ExportabilityReason`? It seems like it could clean this up a bit.

Reply:
Good idea. I was trying to find ways to simplify this logic and didn't consider this alternative. It's updated.